lib/pymedphys/_experimental/streamlit/apps/transfer_check.py

```diff
@@ -70,185 +58,405 @@ def main():
-        index = []
-        for j in dicom_table.iloc[:]["field_label"]:
-            for i in range(len(mosaiq_table)):
-                if mosaiq_table.iloc[i]["field_label"] == j:
-                    index.append(i)
-
-        # Create a list of indices which contain fields not within the RP file
-        remove = []
-        for i in mosaiq_table.iloc[:].index:
-            if i not in index:
-                remove.append(i)
-
-        # Drop all indices in the remove list to get rid of fields irrelevant for this comparison
-        mosaiq_table = mosaiq_table.drop(remove)
-        mosaiq_table = mosaiq_table.sort_index(axis=1)
-        mosaiq_table = mosaiq_table.sort_values(by=["field_label"])
-
-        # Compare values between the two systems and create a new dataframe with the results
-        results = compare_to_mosaiq(dicom_table, mosaiq_table)
-        results = results.transpose()
+def show_field_rx(dicom_table, selected_label):
+    st.write(
+        "**RX**: ",
+        dicom_table[dicom_table["field_label"] == selected_label.values[0]][
+            "rx"
+        ].values[0],
+    )
 
-        # Create a radio selection of prescriptions in mosaiq to choose from for displaying results
-        rx_selection = st.radio("Select RX: ", mosaiq_table.site.unique())
-        rx_fields = mosaiq_table[mosaiq_table["site"] == rx_selection][
-            "field_name"
-        ].values
 
-        # create a radio selection of fields to compare, only fields within selected rx appear as choices
-        field_selection = st.radio("Select field to compare:", rx_fields)
-        selected_label = mosaiq_table[mosaiq_table["field_name"] == field_selection][
-            "field_label"
-        ]
-        dicom_field_selection = dicom_table[
-            dicom_table["field_label"] == selected_label.values[0]
-        ]["field_name"].values[0]
-        st.subheader("Comparison")
+def show_comparison_of_selected_fields(dicom_field_selection, results):
+    dicom_field = str(dicom_field_selection) + "_DICOM"
+    mosaiq_field = str(dicom_field_selection) + "_MOSAIQ"
+    display_results = results[[dicom_field, mosaiq_field]]
 
-        # If a field is selected, write a side by side comparison of the DICOM and MOSAIQ plan information
-        if len(field_selection) != 0:
+    display_results = display_results.drop(
+        ["dob", "first_name", "last_name", "mrn"], axis=0
+    )
 
-            # Write prescription listed in the RP file
-            st.write(
-                "**RX**: ",
-                dicom_table[dicom_table["field_label"] == selected_label.values[0]][
-                    "rx"
-                ].values[0],
-            )
+    display_results = display_results.style.apply(colour_results, axis=1)
+    st.dataframe(display_results.set_precision(2), height=1000)
+
+
+def get_structure_aliases():
+    cwd = os.getcwd().replace("\\", "/")
+    file_path = cwd + "/lib/pymedphys/_experimental/chartchecks/ALIASES.csv"
```

Comment:
cwd won't always be the root of pymedphys 🙂, so this'll bite you later. If you were to create a file within `_experimental/chartchecks` that had the following function that should help:

```python
def get_aliases():
    file_path = pathlib.Path(__file__).parent.joinpath('ALIASES.csv')
    return pd.read_csv(file_path)
```

Reply:
I knew this was going to be a problem whenever I was writing it, but wasn't sure the best way to go about it! Should be taken care of now. 😄  I've left the path for the database set to my personal computer though, since it contains patient mrns and I do not want that committed.